CC tests with high limit format

lutovich opened the pull request:
Commits:

 1. Removed unused package from CC module.
Removes `src/main/java/org/neo4j/consistency/store/paging`

 2. New test high limit format that forces two record units.
Current high limit format uses tow record units only if pointers in the
written record are rather big. This is not very convenient for testing because
it could result in big sparse store files.
This commit adds a new test high limit format that forces records to be split in two record units more often and has a adjustable split threshold.

 3. CC tests for double record unit high limit format.
Currently CC tests are only executed in the neo4j-consistency-check module
which depends only on community neo4j-kernel. This means CC tests only cover
low limit community record format.
This commit adds suite of CC tests to neo4j-enterprise module so they will be
executed with test high limit format that forces records with two record units.

lutovich committed:
Removed unused package from CC module

lutovich committed:
Test high limit format with smaller record sizes

Current high limit format uses two record units only if pointers in the
written record are rather big. This is not very convenient for testing because
it could result in big sparse store files.

This commit adds a new test high limit format that uses smaller node,
relationship and relationship group record sizes. This forces records to be
split in two record units more often.

lutovich committed:
CC tests for double record unit high limit format

Currently CC tests are only executed in the neo4j-consistency-check module
which depends only on community neo4j-kernel. This means CC tests only cover
low limit community record format.

This commit adds suite of CC tests to neo4j-enterprise module so they will be
executed with test high limit format that makes records with two record
units appear more often.

maxsumrall commented on enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/HighLimit.java:
```diff
@@ -42,6 +42,7 @@
  */
 public class HighLimit extends BaseRecordFormats
 {
+    public static final String NAME = "high_limit";
```
Can we also add the same public `NAME` field to LowLimitV3_0.java and remove `COMMUNITY_KEY` from `InternalRecordFormatSelector` and start referencing this new field?

lutovich replied:
Yes, will do

lutovich committed:
Add secondary unit info to #toString()

This commit adds information about if secondary unit is required and it's id
to NodeRecord, RelationshipRecord and RelationshipGroupRecord.

lutovich committed:
Include secondary unit id in #clone()

This commit makes NodeRecord, RelationshipRecord and RelationshipGroupRecord
include id of secondary units when doing #clone().

lutovich committed:
Better assertion when reading secondary unit header

Commit makes assertion an explicit if-check instead of assert. This is check
is needed at runtime to fail fast if inconsistent secondary unit was read from
the store.
